:sparkles: add a script to automatically suggest translations

thomasleveil opened the pull request:
## TLDR

- Run `npm run update-translations`
- review changes
- commit what makes sense

## Motivation

Maintaining locale files is difficult because no dev is fluent in all supported languages. This PR adds a script that uses ChatGPT to provide suggestions for missing translations. It's then up to the developer to choose what translation suggestion is safe to add and commit.
When adding new UI text elements, this would provide at least an AI generated translation that may not be perfect. Humans could then submit PR to improve those new translations.

mckaywrigley committed:
single save

mckaywrigley committed:
mvp

mckaywrigley committed:
prompt adjust

mckaywrigley committed:
readme udpate

mckaywrigley committed:
readme update

mckaywrigley committed:
readme update

mckaywrigley committed:
ocd

mckaywrigley committed:
remove old

mckaywrigley committed:
fix delete error

mckaywrigley committed:
fix deletion bug

mckaywrigley committed:
add demo

SeiyaTakahashi-R committed:
fix: support japanese typing

mckaywrigley committed:
Merge pull request #4 from PeterTakahashi/fix/support-japanese-typing

fix: support japanese typing

mckaywrigley committed:
add markdown

mckaywrigley committed:
readme update

mckaywrigley committed:
code syntax highlights

mckaywrigley committed:
readme update

mckaywrigley committed:
readme update

mckaywrigley committed:
toggle sidebar

mckaywrigley committed:
light/dark code highlight

mckaywrigley committed:
add rename and delete improvements

mckaywrigley committed:
readme update

mckaywrigley committed:
fix minor ui bugs

mckaywrigley committed:
mobile improvements

mckaywrigley committed:
rogue div

mckaywrigley committed:
responsive width

mckaywrigley committed:
increase sidebar btn

mckaywrigley committed:
adjust sidebar btn

mckaywrigley committed:
fix width

mckaywrigley committed:
Update README.md

mckaywrigley committed:
Update README.md

mckaywrigley committed:
fix naming bugs

mckaywrigley committed:
gfm

nauxliu committed:
Update settings button style

mckaywrigley committed:
Merge pull request #8 from nauxliu/settings-button-style

Update settings button style

nauxliu committed:
Ignore new message when the current message is not finished yet

mckaywrigley committed:
Merge pull request #9 from nauxliu/disable-input-when-message-is-streaming

Ignore new message when the current message is not finished yet

mckaywrigley committed:
minor case issue

asolpshinning committed:
Updated ChatInput.tsx text area to autoextend for larger prompt size

mckaywrigley committed:
Update README.md

mckaywrigley committed:
Merge pull request #11 from asolpshinning/patch-1

Updated ChatInput.tsx text area to autoextend for larger prompt size

mckaywrigley committed:
fix syntax

mckaywrigley committed:
add api key

mckaywrigley committed:
fix loader position

mckaywrigley committed:
put it in reverse terry

mckaywrigley committed:
slight input ui tweak

mckaywrigley committed:
change to auto scroll

mckaywrigley committed:
readme update

mckaywrigley committed:
fix 1st convo default name

nauxliu committed:
Correct the improper context being used when it exceeds the limit.

mckaywrigley committed:
Merge pull request #15 from nauxliu/fix-converstaion-context

Correct the improper context being used when it exceeds the limit.

nauxliu committed:
User message should not render as Markdown

mckaywrigley committed:
Merge pull request #16 from nauxliu/fix-user-message-render

User message should not render as Markdown

mckaywrigley committed:
add more mobile ui (#18)

thomasleveil committed:
:whale: add Dockerfile (#17)

Co-authored-by: dev <dev@ptfprod7>

mckaywrigley committed:
Feat/search conversation (#24)

* feat: Add conditional search input for conversation list

* style and ux adjust

---------

Co-authored-by: Tekle Ayele <tekleayele@gmail.com>

sonnylazuardi committed:
Fix safari small screen text input position style (#20)

* Update .gitignore

* Update next.config.js

* Enable GPT-4

* Update index.ts

* Fix logic

* Fix error

* Add set openai key instruction

* Fix menu safari

* Revert changes

* new line

---------

Co-authored-by: Sonny <sonny.lazuardi@grabtaxi.com>

mckaywrigley committed:
readme update

mckaywrigley committed:
Add GPT-4 support (#25)

* mobile ui updates

* fixes sidebar btn

* return if null

* mobile input blur

* handle mobile enter key

* new convo name

* new delete mechanism

* test height

* revert

* change padding

* remove overflow

* check relative

* padding

* done

* retry

* test

* test

* should work now

* test

* test

* more

* max h

* revert

* done

mckaywrigley committed:
readme update

mckaywrigley committed:
api key fix

mckaywrigley committed:
error handling (#27)

nauxliu committed:
Add apple-mobile-web-app meta tags (#29)

dominiksipowicz committed:
remove ENV var from vercel deployment (#30)

mckaywrigley committed:
Token based and model conditional limits (#36)

* use tiktoken for api limit

* model conditional char limits on frontend

* adjust for completion tokens

---------

Co-authored-by: Alan Pogrebinschi <alanpog@gmail.com>

alanpog committed:
wrap user messages (#35)

mckaywrigley committed:
add custom system prompt (#39)

nauxliu committed:
Show latest conversations on the top (#46)

nauxliu committed:
Auto close sidebar when a conversation is selected/created on mobile (#47)

angular-moon committed:
Fixed Search conversations input box was out of bounds (#41)

mckaywrigley committed:
Quocthinhvo/main (#52)

* Change scroll bar style in desktop

* new scrollbar

---------

Co-authored-by: Quốc Thịnh Võ <46624702+quocthinhvo@users.noreply.github.com>

mckaywrigley committed:
Clear conversations (#53)

* Add button "Clear Conversations"

* custom btn

---------

Co-authored-by: Xiangxuan Liu <xiangxuan.liu@rightcapital.com>

mckaywrigley committed:
keep message focus

k11q committed:
enhance ui and fix layout problems (#63)

mckaywrigley committed:
hide rogue scrollbar (#69)

mckaywrigley committed:
fix copy code overlap (#70)

mckaywrigley committed:
add import/export (#71)

mckaywrigley committed:
hotfix

mckaywrigley committed:
hotfix

mckaywrigley committed:
hotfix

mckaywrigley committed:
Feat/add stop conversation button (#76)

* chore add in stop conversation button

* feat: use abort controller

* chore: formatting

---------

Co-authored-by: Simon Holmes <srsholmes@gmail.com>

mckaywrigley committed:
revert

mckaywrigley committed:
okay good

nauxliu committed:
Apply the gradient bg style to input on mobile (#77)

nauxliu committed:
Update the Stop Generating button style (#79)

mckaywrigley committed:
fix light mode

ethereumdegen committed:
improve error msg (#81)

Co-authored-by: apexmfer <apexmfer@gmail.com>

mckaywrigley committed:
add dist

CH4R4F committed:
Name conversations automatically #86 (#90)

CH4R4F committed:
fix style issue after clicking on ClearConversation (#89)

there is a small difference in the spaces of the button after confirming which leads to an unexpected animation in the sidebar settings

patanjalikr13 committed:
added download button, downlaod handler and necessary utility functions (#85)

Co-authored-by: Patanjali Kumar <patanjali@oddup.com>

aekeus committed:
Search within conversation.messages + conversation.name (#91)

mckaywrigley committed:
update readme

mckaywrigley committed:
fix scroll behavior

mckaywrigley committed:
tweak scroll tolerance

mckaywrigley committed:
handle code copy error

aweshchoudhary committed:
SEO Fixed and Added Semantic Html Tags (#98)

mckaywrigley committed:
new load behavior

mckaywrigley committed:
fix rogue messages

jean-jacket committed:
autofocus textarea (#103)

alanpog committed:
include prompt in token count (#104)

mckaywrigley committed:
folders (#108)

* folders

* fixes

* storage fix

mckaywrigley committed:
hotfix import

mckaywrigley committed:
stray log

mckaywrigley committed:
clean history on import

mckaywrigley committed:
Update README.md

Chanzhaoyu committed:
feat: show language name (#113)

bcullman committed:
Update Sidebar Setting elements to be buttons (improve accessibility) (#130)

* update sidebar elements to be buttons so they are keyboard navigatable

* rollback whitespace change

logycon committed:
add k8s config (#117)

mckaywrigley committed:
fix import issue

Commit:
feat: codeblock styling like chatgpt (#132)

mckaywrigley committed:
edit message

mckaywrigley committed:
handle mobile edit

ernestobarrera committed:
Added character count to alert message (#135)

This improvement adds additional information to the alert that is displayed when the user enters more characters than allowed in the rich text element. Now, the alert also shows the number of characters that the user has entered, which helps to provide clearer and more useful feedback to the use

aweshchoudhary committed:
Added Better Closing Sidebar Logic (Close on Background Click) (#122)

mckaywrigley committed:
fix sidebar arrow showing

thomasleveil committed:
Do not rely on user to figure out if server is configured with an api key (#136)

Fixes #105 (and probably also #115)

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

mckaywrigley committed:
fix mobile edit

mckaywrigley committed:
Merge branch 'main' of https://github.com/mckaywrigley/chatbot-ui-pro

Commit:
fix: disable button if messageContent is empty (#137)

mckaywrigley committed:
regenerate button (#138)

mckaywrigley committed:
fix send params

thomasleveil committed:
smoother autoscroll down (#141)

Commit:
fix: change message edit logic (#144)

if the edited message isn't different from the original message, then do nothing.

mckaywrigley committed:
Update README.md

Commit:
feat: Add i18n support for Chinese language (#142)

* feat: Add i18n support for Chinese language

* fix: locale not working in Docker environment

mckaywrigley committed:
hotfix

thomasleveil committed:
Precise error messages (#150)

* Introduce a component to display error messages

* precise error message when api key is invalid

Commit:
feat: Russian language (#148)

* feat: russian (Русский) language

* chore: correct translate

* chore: correct "System prompt" translate

thomasleveil committed:
Locale fr (#153)

* :globe_with_meridians: fr translations

* :coffin: remove unused translation

Anerco committed:
chore: move deps to dev deps (#140)

thomasleveil committed:
:globe_with_meridians: locale de (#154)

Jungley8 committed:
feat: Allow customization of OpenAI host with environment variable (#152)

This commit modifies the OpenAI host configuration to support customization through an environment variable. This change is particularly useful in scenarios where access to the official OpenAI host is restricted or unavailable, allowing users to configure an alternative host for their specific needs.

matias-casal committed:
locale es (#156)

Co-authored-by: Mati <mcasal@bootweb.com.ar>

riande committed:
Add locale sv (Swedish) (#157)

* add locale sv (Swedish)

* add locale sv (Swedish)

thomasleveil committed:
recover from corrupted conversationHistory data (#162)

If an item from conversationHistory is badly corrupted, skip it instead of crashing the app

p-toni committed:
locate pt (#159)

* locate portuguese

* locate portuguese

tulaseerao committed:
Added Telugu language(South Indian) translations (#161)

brianjckim committed:
locale ko (#166)

Co-authored-by: Brian Kim <brian@brianjckim.com>

hyena459 committed:
add_ja_locales (#172)

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

aesadde committed:
fix: update es translations (#167)

Co-authored-by: aesadde <albertosadde@gmail.com>

bcullman committed:
sort the data, not the UI. buttons are now tab-able from the top down (#176)

thejackwu committed:
feat: Message copy button (#171)

* Add copy button

* Fix copy button not copying the entire message

* fix style

* remove prewrap

---------

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

bruce-shi committed:
Latex plugin (#165)

mckaywrigley committed:
hotfix

Commit:
Locale vi (#177)

* Locale vi

* Fix translates layout

hyena459 committed:
Corrected wording (#178)

srsholmes committed:
feat: add in prettier and format code for consistency (#168)

Chanzhaoyu committed:
feat: export json format (#185)

HaithamLeo committed:
added CONTRIBUTING.md file (#188)

Commit:
chore: delete code light theme, like chatgpt (#186)

Commit:
chore: change sidebar font size & style (#191)

* chore: change sidebar font size & style

* chore: create font size style for sidebar

oznav2 committed:
add_Hebrew_locales (#190)

* add_Hebrew_locales

Add Locales of Hebrew

* Update next-i18next.config.js

updating i-18next with Hebrew

Commit:
fix: fix fonts (#194)

* fix: hotfix fonts

* chore: set normal line height

nahid18 committed:
feat: bangla lang support (#195)

Co-authored-by: nahid18 <nahidpatwary50@email.com>

umarhadi committed:
feat: add Bahasa Indonesia support (#198)

Signed-off-by: umarhadi <hi@umarhadi.dev>

tatsuiman committed:
docker build action (#196)

jiayao committed:
Allow switching GPT model in the middle of a conversation (#181)

* Allow switching model in the middle of a conversation

* Hide model selection menu behind a settings button

* rename

* Touch up the settings icon

Commit:
chore: restyle modal with model select (#202)

chore: set normal font size for sidebar
chore: set normal gradient for `ChatInput`

Commit:
Fix unreadable dropdown menu text in dark mode (#204)

* Fix unreadable dropdown menu text in dark mode

* Fix unreadable dropdown menu text in dark mode

Commit:
fix: missing translation (#206)

for translators: need translate `Save & Submit` and `Cancel`

DannyAziz committed:
Add some additional text for API Key (#210)

* feat: API Key Link

* feat: key input placeholder

bcullman committed:
convert conversation icons to buttons (accessibility) (#192)

* update sidebar overflow

* update all clickable icons to buttons

* refactor buttons so they are not inside other buttons

* format doc

* update input background to transparent

* adjust btns size to match #202

* update text size per #202

thomasleveil committed:
Fix rendering performances issues related to scrolling events (#174)

* memoize chat related components

* Avoid re-rendering ChatInput on every message udpate

* change the way the horizontal scrollbar is hidden

* make the scroll event listener passive

* perf(Chat): fix performances issues related to autoscroll

Uses the intersection API to determine autoscroll mode instead of listening for scroll events

* tuning detection of autoscroll

chinleez committed:
Display optimization on iPhone device page (#220)

* Disable Zoom Page

* add height

---------

Co-authored-by: Praise <lizan60@gmail.com>

mckaywrigley committed:
Option for clear all messages (#222)

Co-authored-by: ryanhex53 <ouyang.em@gmail.com>

pujux committed:
fix german wording and switch to informal style (#226)

Commit:
chore: some small improvements (#223)

* chore: stylize error message div
chore: correct styles for sidebar btn
chore: add spinner and replace header "Please wait" on spinner
chore: correct Russian translate
chore: hide clear conversation btn if not conversations
chore: stylize "Need OpenAI key" div

* chore: corrent Russian translate

mckaywrigley committed:
fix regenerate

Commit:
chore: fix styles in "OpenAI Key" div (#228)

chore: do nothing if the name of the renamed dialog contains nothing.
chore: fix some styles
chore: remake conversation settings on mobile

mckaywrigley committed:
Prompts (#229)

SyedMuzamilM committed:
fix: overlap between plus and prompt menu icon (#230)

thomasleveil committed:
fix import  (#242)

* :bug: fix import (#224)

* :bug: fix import of corrupted history

see https://github.com/mckaywrigley/chatbot-ui/issues/224#issuecomment-1486080888

* add the run-test-suite github action

bcullman committed:
convert folders to buttons & folder icons to buttons (accessibility)  (#237)

* tabbable folders

* fix spacing

bcullman committed:
update prompts to be tabbable (#241)

Chanzhaoyu committed:
feat: empty state add icons (#248)

SyedMuzamilM committed:
fix: prompt sidebar not showing on small devices (#232)

* fix: overlap between plus and prompt menu icon

* fix: prompt sidebar not showing on small devices

bcullman committed:
make all chat area components tabbable (accessibility) (#246)

* make all chat area components tabbable

* align message role description

* remove inline styles on icons

* remove inline styles on icons

Chanzhaoyu committed:
chore: update chinese locales (#247)

* chore: update chinese locales

* chore: update locales

liby committed:
fix: resolve Enter event conflict during CJK IME (#253)

* fix: resolve Enter event conflict during CJK IME

* add

---------

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

mckaywrigley committed:
scroll btn (#256)

* feat: added scroll down button

when the use scrolls up on the chat a button will appear on the bottom right side of the screen. It will smoothly scroll down to the bottom of the chat when the button was pressed.

* remove env

---------

Co-authored-by: dasunNimantha <dasun4@pm.me>

mckaywrigley committed:
Update README.md

srsholmes committed:
chore: migrate to vitest for 10x faster tests (#257)

* chore migrate to vitest

* chore: cleanup jest stuff

* chore: install the coverage dep

SyedMuzamilM committed:
fix: layout shifting on small devices (#258)

* fix: overlap between plus and prompt menu icon

* fix: prompt sidebar not showing on small devices

* fix: layout shifting on small devices when there is a code block

dasunNimantha committed:
style: changed scroll icon size via size attribute as suggested (#263)

SyedMuzamilM committed:
fix: website crash when typing / followed with non prompt name (#262)

* fix: froward slash crash

* added the rounded corners when editing folder/chat

Commit:
doc: update doc about Docker (#264)

maybe fix #259

ernestobarrera committed:
New JSON translation file for Spanish (#270)

Translation to Spanish of the prompts custom template bar

DjTer3a committed:
Adding arabic translation (#274)

thecrazybob committed:
Fix invalid Tailwind property (#275)

dasunNimantha committed:
chore: added Sinhala language locales (#276)

thomasleveil committed:
:sparkles: feat: add DEFAULT_MODEL environment variable (#280)

* :sparkles: feat: add DEFAULT_MODEL environment variable

* set the model maxLength setting in the models definition

* set the model tokenLimit setting in the models definition

Chanzhaoyu committed:
fix: translation omission (#281)

mckaywrigley committed:
Update README.md

mckaywrigley committed:
change messaging

dasunNimantha committed:
fix: In light mode, the arrow in the "scroll to bottom" icon is difficult to see #299 (#304)

ryanhex53 committed:
return detail error from api call. (#295)

burke committed:
Allow specifying OpenAI-Organization header: (#313)

See https://platform.openai.com/docs/api-reference/introduction

> For users who belong to multiple organizations, you can pass a header
> to specify which organization is used for an API request. Usage from
> these API requests will count against the specified organization's
> subscription quota.

SyedMuzamilM committed:
fix: small ui issue (#311)

* fix: unnecessary lines before the chats or prompts

Removed unncessary lines if there are no folders before prompts and
chats.

* added the light mode for scroll down button

MatriQ committed:
fix: properly trigger prompt modal close (#309)

Co-authored-by: MatriQi <matri@aifi.io>

liby committed:
feat: support custom default system prompt (#285)

mckaywrigley committed:
change default prompt in example

mckaywrigley committed:
fix nasty bug

mckaywrigley committed:
update demo

rudolfolah committed:
Feature: add link to openai account usage (#319)

* Update ModelSelect.tsx to include link to account usage

It would be helpful to be able to click a link to see the account usage and billing, and the model selection is related to the pricing.

* updated styling and added icon for external link

* fixed missing import

brijeshb42 committed:
Autofocus input when after clicking OpenAI API Key in sidebar. (#324)

* Update Key.tsx

Autofocus input after its visible.

* Update Key.tsx

colindotfun committed:
fix: label layout shift (#338)

Commit:
fix: prompts list height (#340)

danielkhoo committed:
Fix icon color (#344)

mckaywrigley committed:
fix spacing

riande committed:
locale improvements (#351)

aaliboyev committed:
fix for openai token limit error (#350)

superman66 committed:
feat: support import and export with prompts (#330)

* feat: support import and export prompts

* test: update importExports.test.ts

* Delete .gitpod.yml

mckaywrigley committed:
change to "data"

Jdban committed:
Adds a loading spinner that replaces send icon while result is generated (#329)

Jdban committed:
add react-hot-toast and surface OpenAI API errors to users (#328)

SyedMuzamilM committed:
fix: some ui issues (#346)

* fix: scroll button not visible in light mode

* fix: sidebar when there is a folder

When a folder is added in the sidebar and there are less items scroll
bar appears. This simple change fixes that behaviour

* fix: small devices regerate/Stop button below input

Below 768px Stop Genrating and Regerate Button remains hidden behind the
input. This is the fix for that

bcullman committed:
remove dupe main (#355)

bcullman committed:
remove unused code (#356)

mckaywrigley committed:
change output limit

mckaywrigley committed:
Merge branch 'main' of https://github.com/mckaywrigley/chatbot-ui-pro

spctechdev committed:
Locale it (#363)

* Add files via upload

* Update next-i18next.config.js

* Update chat.json

mckaywrigley committed:
revert

mckaywrigley committed:
Merge branch 'main' of https://github.com/mckaywrigley/chatbot-ui-pro

thomasleveil committed:
:sparkles: add a script to automatically suggest translations

Run `npm run update-translations`

alanpog committed:
tokenLimit now can be closer to real model limit. (#366)

thomasleveil committed:
Fix translations import export (#376)

* fix "Import data" case

* fix "Export data" translations

thomasleveil committed:
:notes document (#373)

jpzhangvincent committed:
update chat role icon (#378)

Commit:
Update Vietnamese (#385)

spctechdev committed:
Fix previous IT locale (#374)

* Update chat.json

* Update promptbar.json

* Update sidebar.json

* Update sidebar.json

* Update sidebar.json

oznav2 committed:
Update promptbar.json (#389)

fix : added missing translation

srsholmes commented:
![silicon-valley](https://user-images.githubusercontent.com/3579905/229463947-b23cc3fa-a60f-4847-a73c-fb574e074570.gif)

What an awesome use of AI. ❤️ Love this.  

Fantastic work @thomasleveil.

ivanfioravanti commented on scripts/update-translations.ts:
```diff
@@ -0,0 +1,91 @@
+        role: 'system',
+        content: `You are a translator and know all languages. 
+You are tasked with maintaining an application locale translation files expressed in JSON.
+The JSON content is composed of keys representing English label that must not change, and values that represent the tranlation of those labels.
```
tranlation instead of translation

thomasleveil replied:
Good catch ! fixed

mckaywrigley committed:
push (#414)

mckaywrigley committed:
Google fixes (#415)

* push

* fixes

mckaywrigley committed:
brevity

mckaywrigley committed:
spelling

thomasleveil commented:
rebased

itbm committed:
Hide API Key and Plugin Key buttons if set server side (#423)

thomasleveil committed:
update README regarding GOOGLE_API_KEY and GOOGLE_CSE_ID (#422)

thomasleveil committed:
Fix 416 copy button on code block (#427)

* update README regarding GOOGLE_API_KEY and GOOGLE_CSE_ID

* :bug: fix copy button not showing up on code block with no recognized language

fixes #416

thomasleveil committed:
update CONTRIBUTING to explain how to generate translations

mckaywrigley committed:
MAJOR REFACTOR (#494)

* move index to home folder, create state and context files and barrell folder

* Sanity Check Commit:  reducer added to home.tsx manual QA all working

* WIP: promptBar

* fix missing json parse on folders and prompts

* split context and add promptbar context

* add context to nested prompt componets and componetize Folder componet

* remove log

* Create buttons folder and componetize sidebar action button

* tidy up prompt handlers

* componetized sidebar

* added back chatbar componet to left side sidebar

* monster commit: Componetized the common code between chatbar and promptbar into new componet Sidebar and added context to both bars

* add useFetch service

* added prettier import sort to keep imports ordered and easier to indentify

* added react query and useFetch to work with RQ

* added apiService, errorService and reactQuery

* add callback and tidy up error service

* refactor chat and child componets to useContext

* fix extra calls and bad calls to mel endpoint

* minor import cleanup

---------

Co-authored-by: jc.durbin <jc.durbin@ardanis.com>

akashkriplani committed:
fix parameter of length to type "number" (#492)

dotneet committed:
fix google plugin bug due to URI encoding issue. (#469)

* fix: uri encoding

* fix: error handling

MKTHINGS committed:
Add Romanian translation (#468)

* Create Romanian chat.json

I have created Romanian translation of the chat.json.

* Add Romanian translation

I have created Romanian translations.

* Update next-i18next.config.js

Added ro locale

---------

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

regisrex committed:
Update demo image (#459)

thomasleveil committed:
document OPENAI_API_HOST and OPENAI_ORGANIZATION (#429)

dornfeld-capital committed:
improve plugin hotkey support (#426)

itbm committed:
Add support for Azure OpenAI (#495)

mckaywrigley committed:
log removal

jony4 committed:
fix: issues/498 (#499)

thomasleveil commented:
rebased
